scripts/jenkins/ardana/gerrit/gerrit_handle_event.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python
+
+import argparse
+import os
+import sys
+
+sys.path.append(os.path.dirname(__file__))
+from gerrit import GerritChange, GerritChangeSet  # noqa: E402
+from gerrit_merge import gerrit_merge  # noqa: E402
+from gerrit_review import gerrit_review  # noqa: E402
+
+
+def get_submittable_references(merged_change):
+    """
+    Get all submittable changes that directly depend on the supplied change
+    (have an explicit Depends-On referencing the indicated merged Gerrit
+    change and targeting the same branch, or that are submitted together
+    and directly on top of it).
+
+    :param merged_change: GerritChange object representing a merged change
+    :return: list of GerritChange object representing submittable changes,
+    referencing the supplied merged change as a direct dependency, and which
+    could be merged themselves
+    """
+
+    changeset = GerritChangeSet(
+        'is:open',
+        'is:mergeable',
+        'branch:{}'.format(merged_change.branch),
+        'label:Code-Review=2',
+        'label:Verified=2',
+        'label:Workflow=1',
+        '-label:Verified=-1',
+        '-label:Code-Review=-2'
+    )
+
+    references = []
+    for change in changeset.changes():
+        if change.submittable and change.depends_on_directly(merged_change):
+            references.append(change)
+
+    return references
+
+
+def get_stale_references(updated_change):
+    """
+    Get all changes that directly or indirectly (i.e. through a chain of
+    Depends-On commit message indicators) depend on the indicated updated
+    Gerrit change and targeting the same branch.
+
+    :param updated_change: GerritChange object representing a change
+    that has been updated (e.g. for which a new patchset was published)
+    :return: list of GerritChange object representing changes directly or
+    indirectly referencing the supplied updated change as a dependency,
+    which need to be re-verified
+    """
+    changeset = GerritChangeSet(
+        'is:open',
+        'branch:{}'.format(updated_change.branch),
+        'message:"Depends-On:"'
+    )
+
+    # All open changes in the dependency tree will be added to this list
+    references = [updated_change]
+    for change in references:
+        for open_change in changeset.changes():
+            if open_change not in references and \
+               open_change.depends_on_directly(change):
+                references.append(open_change)
+
+    # Remove the triggering change
+    references.pop(0)
+    return references
+
+
+def handle_change_merged(change_no):
+    """
+    After a Gerrit change is merged, track down all the other open and
+    submittable changes with a Depends-On marker pointing to the merged
```

Comment:
This method should also track down child commits.

Reply:
Done